aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -259,13 +261,23 @@ private boolean isGlueIcebergTable(Table table) {
   public boolean dropTable(TableIdentifier identifier, boolean purge) {
     try {
       TableOperations ops = newTableOps(identifier);
-      TableMetadata lastMetadata = ops.current();
+      TableMetadata lastMetadata;
```

Comment:
discussed offline, loading table metadata still feels hacky to me. We can get Glue table and check the metadata location to determine if the table is already purged or not.